scraper/scraper.py

```diff
@@ -869,17 +597,104 @@ def login(email, password):
         exit(1)
 
 
-# -----------------------------------------------------------------------------
-# -----------------------------------------------------------------------------
+def scrap_profile(ids):
+    folder = os.path.join(os.getcwd(), "data")
+    utils.create_folder(folder)
+    os.chdir(folder)
 
+    # execute for all profiles given in input.txt file
+    for user_id in ids:
 
-def scraper(**kwargs):
-    with open("credentials.yaml", "r") as ymlfile:
-        cfg = yaml.safe_load(stream=ymlfile)
+        driver.get(user_id)
+        url = driver.current_url
+        user_id = create_original_link(url)
 
-    if ("password" not in cfg) or ("email" not in cfg):
-        print("Your email or password is missing. Kindly write them in credentials.txt")
-        exit(1)
+        print("\nScraping:", user_id)
+
+        try:
+            target_dir = os.path.join(folder, user_id.split("/")[-1])
+            utils.create_folder(target_dir)
+            print(target_dir)
+            os.chdir(target_dir)
+        except Exception:
+            print("Some error occurred in creating the profile directory.")
+            continue
+
+
+        to_scrap = ['Friends', 'Photos', 'Videos', 'About', 'Posts']
+        for item in to_scrap:
+            print("----------------------------------------")
+            print("Scraping {}..".format(item))
+            
+            if item == "Posts":
+                scan_list = [None]
+            elif item == "About":
+                scan_list = [None] * 7
+            else:
+                scan_list = params[item]['scan_list']
+
+            section = params[item]['section']
+            elements_path = params[item]['elements_path']
+            file_names = params[item]['file_names']
+            save_status = params[item]['save_status']
+
+            scrape_data(user_id, scan_list, section, elements_path, save_status, file_names)
+            
+            print("{} Done!".format(item))
+
+    print("\nProcess Completed.")
+    os.chdir("../..")
+
+    return
+
+
+def scrape_data(user_id, scan_list, section, elements_path, save_status, file_names):
+    """Given some parameters, this function can scrap friends/photos/videos/about/posts(statuses) of a profile"""
+    page = []
+
+    if save_status == 4:
+        page.append(user_id)
+
+    page += [user_id + s for s in section]
+
+    for i, _ in enumerate(scan_list):
+        try:
+            driver.get(page[i])
+            assert "facebook" in driver.title
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/harismuneer/Ultimate-Facebook-Scraper/pullRequest?prid=5254804)